docs/tutorials/constraints.rst

```diff
@@ -0,0 +1,255 @@
+
+    # Read in the table of targets
+    from astropy.io import ascii
+    target_table = ascii.read('targets.txt')
+
+    # Create astroplan.FixedTarget objects for each one in the table
+    from astropy.coordinates import SkyCoord
+    import astropy.units as u
+    targets = [FixedTarget(coord=SkyCoord(ra=ra*u.deg, dec=dec*u.deg), name=name)
+               for name, ra, dec in target_table]
+
+In the above example, you may have noticed that constraints are assembled by
+making a list of calls to the initializers for classes like
+`astroplan.AltitudeConstraint` and `astroplan.AirmassConstraint`. Each of those
+constraint classes is subclassed from the abstract `Constraint` class, and the
+custom constraint that we're going to write must be as well.
+
+In this example, let's design our constraint to ensure that all targets must be
+within some angular separation from Vega – we'll call it
+`VegaSeparationConstraint`. Two methods, `__init__` and `compute_constraint`
+must be written for our constraint to work:
+
+* The `__init__` method will accept the minimum and maximum acceptable separations
+  a target could have from Vega.
+
+* We'll also define a method `compute_constraints` which takes three arguments:
+  an array of times to test, an `astroplan.Observer` object, and one or a list
+  of `astroplan.FixedTarget` objects. `compute_constraints` will return a matrix
+  of booleans that describe whether or not each target meets the constraints.
+  The super class `Constraint` has a `__call__` method which will run your
+  custom class's `compute_constraints` method when you check if a target is
+  observable using `astroplan.is_observable` or
+  `astroplan.is_always_observable`.
+
+Here's our `VegaSeparationConstraint` implementation::
+
+    from astroplan import Constraint, is_observable
+    from astropy.coordinates import Angle
+
+    class VegaSeparationConstraint(Constraint):
+        """
+        Constraint the separation from Vega
+        """
+        def __init__(self, min=None, max=None):
+            """
+            min : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            max : `~astropy.units.Quantity` or `None` (optional)
+                Minimum acceptable separation between Vega and target. `None`
+                indicates no limit.
+            """
+            self.min = min
+            self.max = max
+
+        def compute_constraint(self, times, observer, targets):
+
+            # Vega's position is essentially unchanging, but if it were to be a
+            # moving target, we would need an array of coordinates for Vega as a
+            # function of time. Here we'll simulate that behavior with multiple
+            # copies of the Vega coordinate
+            vega = SkyCoord(ra=279.23473479*u.deg, dec=38.78368896*u.deg)
+            vega = SkyCoord(len(times)*[vega])
```

Comment:
Actually, I think you don't need to do this.  If instead you define it as:
```
vega = SkyCoord(ra=[279.23473479]*u.deg, dec=[38.78368896]*u.deg)
```
ordinary numpy broadcasting happens and all is fine without the extra computational expensive of duplicating `vega` a bunch

In the future, it might be better to try to change the observability calcs to have something like ``numpy.atleast_1d`` in them to allow even the scalar case.  But for now I think just the small change I said above is fine.